src/sentry/incidents/endpoints/serializers/alert_rule.py

```diff
@@ -436,7 +445,10 @@ def get_attrs(
                         "id": item_id,
                         "issue_rule": isinstance(item, Rule),
                         "metric_rule": isinstance(item, AlertRule),
-                        "uptime_rule": isinstance(item, ProjectUptimeSubscription),
+                        "uptime_rule": (
+                            isinstance(item, Detector)
+                            and item.type == GROUP_TYPE_UPTIME_DOMAIN_CHECK_FAILURE
+                        ),
                         "crons_rule": isinstance(item, Monitor),
```

Comment:
This is kind of a weird structure... shouldn't this just be `item_type: type(item)`?

I know this isn't your code, but maybe rather than adding to this pattern, just add both `type(item)` and `item.type` here, so that others don't have to keep adding this weird logic

Reply:
What do you mean exactly?

Reply:
oh this is for that legacy combined alert rules listing page that I think will go away with detectors right